db-tests/src/test/java/gov/nasa/jpl/aerie/database/PlanCollaborationTests.java

```diff
@@ -3628,33 +3614,23 @@ class PlanMigrationTests{
     void checkFailsForNonexistentPlansOrModels() throws SQLException {
       final int existingPlanId = merlinHelper.insertPlan(missionModelId);
 
-      try{
-        migratePlanToModel(-1, missionModelId);
-        fail();
-      }
-      catch (SQLException sqEx){
-        if(!sqEx.getMessage().contains("Plan -1 does not exist"))
-          throw sqEx;
-      }
-
-      try{
-        migratePlanToModel(existingPlanId, -1);
-        fail();
-      }
-      catch (SQLException sqEx){
-        if(!sqEx.getMessage().contains("Model -1 does not exist"))
-          throw sqEx;
-      }
+      assertThrows(
+          SQLException.class,
+          () -> migratePlanToModel(-1, missionModelId),
+          "Plan -1 does not exist"
```

Comment:
This is the message the test fails with if the code doesn't throw a SQLException, and is not equivalent to the `if` check from before. The following code is:

```java
final var sqEx =  assertThrows(SQLException.class, () -> migratePlanToModel(-1, missionModelId));
if(!sqEx.getMessage().contains("Model -1 does not exist")) {
   throw sqEx;
}
```

Reply:
fixed in [4eb45a1](https://github.com/NASA-AMMOS/aerie/pull/1665/commits/4eb45a14c2bb86afa51cb26e3eb8284cfb7ae624)

Reply:
Ookay, this is back to passing and ready for another look @Mythicaeda, TL;DR good catch and sorry for the confusion.

After trying a few things with @mattdailis I think my preferred pattern here is something like:
```
assertTrue(
  sqlEx.getMessage().contains("Plan -1 does not exist"), 
  "bad error message, got " + sqlEx.getMessage()
);
```
since it is still an assertion failure but preserves the message so you can see what was unexpected? This might be a good thing to have a util for (`assertThrowsWithMessage`), but I think we've gone deep enough on this one for now :)

Reply:
That's fine, although some context as to why the reason that the general pattern has been `if(!ex.contains(message)) { throw ex }` (plus or minus checking the SQL exception code, if a custom one has been set):

If there is an exception of any kind during a DB request, it will come back as a SQLException. This includes exceptions due to DB connection issues. The purpose of the "if-not-throw" pattern is to have the test correctly end in an exception if one was thrown, similar to how currently in the E2E tests, if there's an IOException while connecting to the Gateway, the test is ended via an IOException.